ui/src/app/pages/workspace/workspace-about.tsx

```diff
@@ -226,7 +237,7 @@ export const WorkspaceAbout = fp.flow(withUserProfile(), withUrlParams())
                   data-test-id='workspaceShareButton'
                   onClick={() => this.setState({sharing: true})}>Share</Button>
         </div>
-        {workspaceUserRoles.length > 0 ?
+        {(!loadingWorkspaceUserRoles && workspaceUserRoles.length > 0) ?
```

Comment:
If I'm reading this right, this fragment will only render when there are roles to show, regardless of whether the user roles are being loaded. I wonder -- could we set `workspaceUserRoles = []` in `onShare()` or `loadUserRoles()`, and delete the `loadingWorkspaceUserRoles` variable?

Reply:
well technically it will be not be shown when roles are not "loading" AND there are roles to show, because we want to show a spinner in case of reload even when we already have a set of roles (rather than have the change suddenly pop up when the api call finishes).  I'm just personally not a fan of "erasing" a variable so to speak when it gets reloaded - it feels inaccurate.  because in the interim while we are loading the updated user roles the current roles still exist, so to unset them even just as logic for displaying a spinner feels semantically wrong to me.